parsec/core/fs/workspacefs/versioning_helpers.py

```diff
@@ -14,24 +14,51 @@
 )
 
 
+SYNC_GUESSED_TIME_FRAME = 30
+
+
+class TimestampBoundedEntry(NamedTuple):
+    id: EntryID
+    version: int
+    early: Pendulum
+    late: Pendulum
+
+
+class TimestampBoundedData(NamedTuple):
+    device_id: DeviceID
+    updated: Pendulum
+    is_dir: bool
+    size: int
+    source: FsPath
+    destination: FsPath
+
+
+class ManifestData(NamedTuple):
+    device_id: DeviceID
+    updated: Pendulum
+    is_dir: bool
+    size: int
+
+
+class ManifestDataAndPaths(NamedTuple):
+    data: ManifestData
+    source: FsPath
+    destination: FsPath
+
+
 async def list_versions(
-    workspacefs, path: FsPath
-) -> Dict[
-    Tuple[EntryID, int, Pendulum, Pendulum],
-    Tuple[Tuple[DeviceID, Pendulum, bool, int], FsPath, FsPath],
-]:
+    workspacefs, path: FsPath, remove_supposed_minimal_sync: bool = True
+) -> Dict[TimestampBoundedEntry, TimestampBoundedData]:
```

Comment:
En regardant où est utilisé `list_versions` (enfin plutôt `WorkspaceFS.versions`), je ne comprends pas pourquoi on a besoin de retourner quelque chose de si compliqué:
- dans `ts_ws_dialog.py`, on a juste besoin du timestamp de la v0 du workspace manifest (on pourrait obtenir ça en faisant juste un `workspacefs.remote_loader.list_versions` sur l'id du workspace)
- dans `file_history_dialog.py`, on itère sur les clé et valeurs donc autant retourner une liste d'instances d'une seule structure